### ISSUE TITLE
Deploy failing at the slurm installation step

### ISSUE THREAD

**Issue by aktech** [nebari-dev/nebari-slurm#134]
I am trying to deploy qhub-hpc on a set on three nodes (one master and two worker nodes), this is what happens at the following stage

## Task

```
TASK [slurm : Install Slurmd execution host packages] **************************************************************************************************************************************************************************************************************************
```

## Error
```
fatal: [hpc02-test]: FAILED! => {
    "cache_update_time": 1706708780,
    "cache_updated": false,
    "changed": false,
    "invocation": {
        "module_args": {
            "allow_unauthenticated": false,
            "autoclean": false,
            "autoremove": false,
            "cache_valid_time": 3600,
            "deb": null,
            "default_release": null,
            "dpkg_options": "force-confdef,force-confold",
            "force": false,
            "force_apt_get": false,
            "install_recommends": null,
            "name": "slurmd",
            "only_upgrade": false,
            "package": [
                "slurmd"
            ],
            "policy_rc_d": null,
            "purge": false,
            "state": "present",
            "update_cache": null,
            "update_cache_retries": 5,
            "update_cache_retry_max_delay": 12,
            "upgrade": null
        }
    },
    "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"      install 'slurmd'' failed: E: Sub-process /usr/bin/dpkg returned an error code (1)\n",
    "rc": 100,
    "stderr": "E: Sub-process /usr/bin/dpkg returned an error code (1)\n",
    "stderr_lines": [
        "E: Sub-process /usr/bin/dpkg returned an error code (1)"
    ],
    "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nThe following packages were automatically installed and are no longer required:\n  libflashrom1 libftdi1-2\nUse 'sudo apt autoremove' to remove them.\nThe following NEW packages will be installed:\n  slurmd\n0 upgraded, 1 newly installed, 0 to remove and 13 not upgraded.\nNeed to get 209 kB of archives.\nAfter this operation, 626 kB of additional disk space will be used.\nGet:1 http://us.archive.ubuntu.com/ubuntu jammy/universe amd64 slurmd amd64 21.08.5-2ubuntu1 [209 kB]\nFetched 209 kB in 0s (935 kB/s)\nSelecting previously unse
lected package slurmd.\r\n(Reading database ... \r(Reading database ... 5%\r(Reading database ... 10%\r(Reading database ... 15%\r(Reading database ... 20%\r(Reading database ... 25%\r(Reading database ... 30%\r(Reading database ... 35%\r(Reading database ... 40%\r(Reading database ... 45%\r(Reading database ... 50%\r(Reading database ... 55%\r(Reading database ... 60%\r(Reading database ... 65%\r(Reading database ... 70%\r(Reading database ... 75%\r(Reading database ... 80%\r(Reading database ... 85%\r(Reading database ... 90%\r(Reading database ... 95%\r(Reading database ... 100%\r(Reading database ... 112656 files and directo
ries currently installed.)\r\nPreparing to unpack .../slurmd_21.08.5-2ubuntu1_amd64.deb ...\r\nUnpacking slurmd (21.08.5-2ubuntu1) ...\r\nSetting up slurmd (21.08.5-2ubuntu1) ...\r\nupdate-alternatives: using /usr/sbin/slurmd-wlm to provide /usr/sbin/slurmd (slurmd) in auto mode\r\nJob for slurmd.service failed because the control process exited with error code.\r\nSee \"systemctl status slurmd.service\" and \"journalctl -xeu slurmd.service\" for details.\r\ninvoke-rc.d: initscript slurmd, action \"start\" failed.\r\n\u001b[0;1;31mx\u001b[0m slurmd.service - Slurm node daemon\r\n     Loaded: loaded (\u001b]8;;file://jhub-compute
-01/etc/systemd/system/slurmd.service\u0007/etc/systemd/system/slurmd.service\u001b]8;;\u0007; disabled; vendor preset: enabled)\r\n     Active: \u001b[0;1;31mfailed\u001b[0m (Result: exit-code) since Wed 2024-01-31 07:47:54 CST; 33ms ago\r\n    Process: 104929 ExecStart=/usr/sbin/slurmd \u001b[0;1;31m(code=exited, status=1/FAILURE)\u001b[0m\r\n        CPU: 145ms\r\n\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]: slurmd: error: fetch_config: DNS SRV lookup failed\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]: slurmd: error: _establish_configuration: failed to load configs\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]
: slurmd: error: slurmd initialization failed\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: resolve_ctls_from_dns_srv: res_nsearch error: Unknown host\u001b[0m\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: fetch_config: DNS SRV lookup failed\u001b[0m\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: _establish_configuration: failed to load configs\u001b[0m\r\nJan 31 07:47:54 jhub-compute-01 systemd[1]: \u001b[0;1;39m\u001b[0;1;31m\u001b[0;1;39mslurmd.service: Control proces
s exited, code=exited, status=1/FAILURE\u001b[0m\r\nJan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: slurmd initialization failed\u001b[0m\r\nJan 31 07:47:54 jhub-compute-01 systemd[1]: \u001b[0;1;38;5;185m\u001b[0;1;39m\u001b[0;1;38;5;185mslurmd.service: Failed with result 'exit-code'.\u001b[0m\r\nJan 31 07:47:54 jhub-compute-01 systemd[1]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31mFailed to start Slurm node daemon.\u001b[0m\r\ndpkg: error processing package slurmd (--configure):\r\n installed slurmd package post-installation script subprocess returned error exit status 1\r\nProce
ssing triggers for man-db (2.10.2-1) ...\r\nErrors were encountered while processing:\r\n slurmd\r\nneedrestart is being skipped since dpkg has failed\n",
    "stdout_lines": [
        "Reading package lists...",
        "Building dependency tree...",
        "Reading state information...",
        "The following packages were automatically installed and are no longer required:",
        "  libflashrom1 libftdi1-2",
        "Use 'sudo apt autoremove' to remove them.",
        "The following NEW packages will be installed:",
        "  slurmd",
        "0 upgraded, 1 newly installed, 0 to remove and 13 not upgraded.",
        "Need to get 209 kB of archives.",
        "After this operation, 626 kB of additional disk space will be used.",
        "Get:1 http://us.archive.ubuntu.com/ubuntu jammy/universe amd64 slurmd amd64 21.08.5-2ubuntu1 [209 kB]",
        "Fetched 209 kB in 0s (935 kB/s)",
        "Selecting previously unselected package slurmd.",
        "(Reading database ... ",
        "(Reading database ... 5%",
        "(Reading database ... 10%",
        "(Reading database ... 15%",
        "(Reading database ... 20%",
        "(Reading database ... 25%",
        "(Reading database ... 30%",
        "(Reading database ... 35%",
        "(Reading database ... 40%",
        "(Reading database ... 45%",
        "(Reading database ... 50%",
        "(Reading database ... 55%",
        "(Reading database ... 60%",
        "(Reading database ... 65%",
        "(Reading database ... 70%",
        "(Reading database ... 75%",
        "(Reading database ... 80%",
        "(Reading database ... 85%",
        "(Reading database ... 90%",
        "(Reading database ... 95%",
        "(Reading database ... 100%",
        "(Reading database ... 112656 files and directories currently installed.)",
        "Preparing to unpack .../slurmd_21.08.5-2ubuntu1_amd64.deb ...",
        "Unpacking slurmd (21.08.5-2ubuntu1) ...",
        "Setting up slurmd (21.08.5-2ubuntu1) ...",
        "update-alternatives: using /usr/sbin/slurmd-wlm to provide /usr/sbin/slurmd (slurmd) in auto mode",
        "Job for slurmd.service failed because the control process exited with error code.",
        "See \"systemctl status slurmd.service\" and \"journalctl -xeu slurmd.service\" for details.",
        "invoke-rc.d: initscript slurmd, action \"start\" failed.",
        "\u001b[0;1;31mx\u001b[0m slurmd.service - Slurm node daemon",
        "     Loaded: loaded (\u001b]8;;file://jhub-compute-01/etc/systemd/system/slurmd.service\u0007/etc/systemd/system/slurmd.service\u001b]8;;\u0007; disabled; vendor preset: enabled)",
        "     Active: \u001b[0;1;31mfailed\u001b[0m (Result: exit-code) since Wed 2024-01-31 07:47:54 CST; 33ms ago",
        "    Process: 104929 ExecStart=/usr/sbin/slurmd \u001b[0;1;31m(code=exited, status=1/FAILURE)\u001b[0m",
        "        CPU: 145ms",
        "",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: slurmd: error: fetch_config: DNS SRV lookup failed",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: slurmd: error: _establish_configuration: failed to load configs",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: slurmd: error: slurmd initialization failed",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: resolve_ctls_from_dns_srv: res_nsearch error: Unknown host\u001b[0m",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: fetch_config: DNS SRV lookup failed\u001b[0m",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: _establish_configuration: failed to load configs\u001b[0m",
        "Jan 31 07:47:54 jhub-compute-01 systemd[1]: \u001b[0;1;39m\u001b[0;1;31m\u001b[0;1;39mslurmd.service: Control process exited, code=exited, status=1/FAILURE\u001b[0m",
        "Jan 31 07:47:54 jhub-compute-01 slurmd[104929]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31merror: slurmd initialization failed\u001b[0m",
        "Jan 31 07:47:54 jhub-compute-01 systemd[1]: \u001b[0;1;38;5;185m\u001b[0;1;39m\u001b[0;1;38;5;185mslurmd.service: Failed with result 'exit-code'.\u001b[0m",
        "Jan 31 07:47:54 jhub-compute-01 systemd[1]: \u001b[0;1;31m\u001b[0;1;39m\u001b[0;1;31mFailed to start Slurm node daemon.\u001b[0m",
        "dpkg: error processing package slurmd (--configure):",
        " installed slurmd package post-installation script subprocess returned error exit status 1",
        "Processing triggers for man-db (2.10.2-1) ...",
        "Errors were encountered while processing:",
        " slurmd",
        "needrestart is being skipped since dpkg has failed"
    ]
}

RUNNING HANDLER [mysql : restart services mysql] *******************************************************************************************************************************************************************************************************************************
```

## Versions

```
$ cat /etc/*release

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by coolrazor007** [comment]
I know on the README.md they say only Ubuntu 18.04 and Ubuntu 20.04, but I have an interest in this too on 22.04.
From the output it looks like it is either a DNS issue or a package manager issue installing slurmd.  Perhaps test manually installing slurmd on 22.04 by hand to see what is needed.  Then we take that info to update the Ansible playbooks or other relevant code.

**Comment by aktech** [comment]
This is fixed now and ubuntu 22.04 is the supported version.
